vllm/v1/worker/gpu_worker.py

```diff
@@ -398,12 +398,27 @@ def compile_or_warm_up_model(self) -> None:
         # but users still want to compile for better performance,
         # e.g. for the max-num-batched token size in chunked prefill.
         warmup_sizes = self.vllm_config.compilation_config.compile_sizes.copy()
-        if not self.model_config.enforce_eager:
+
+        if (
+            not self.model_config.enforce_eager
+            or self.compilation_config.cudagraph_mode == CUDAGraphMode.NONE
+        ):
             warmup_sizes = [
                 x
                 for x in warmup_sizes
                 if x not in self.vllm_config.compilation_config.cudagraph_capture_sizes
             ]
+        compile_ranges = self.vllm_config.compilation_config.get_compile_ranges()
+
+        # For each compile_range, if none of the batch sizes
+        # in warmup_sizes or cudagraph_capture_sizes are in the range,
+        # add the start of the range to ensure compilation/warmup.
```

Comment:
Can we do end-1 instead of start?

Reply:
In the current impl we only warm up the largest size in the range, which triggers the compilation for the range